### PR TITLE
[BugFix] concurrently rewrite jit after prepare/open original expr in filter

### PR DESCRIPTION
**Pull request by fzhedu**
## What I'm doing:

Fixes https://github.com/StarRocks/StarRocksTest/issues/6261

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by stdpain — DISMISSED** [review]

File: test/conf/sr.conf [hunk 1]
```diff
@@ -43,3 +43,5 @@ aws_region =
 aws_assume_role = 
 aws_sts_region = 
 aws_sts_endpoint =
+
+udf_url = http://starrocks-thirdparty.oss-cn-zhangjiakou.aliyuncs.com
```
Comment (silverbullet233) [inline test/conf/sr.conf:47]: should remote value part

Comment (fzhedu) [reply test/conf/sr.conf:47]: has check with admit, it is ok

**Review by stdpain — DISMISSED** [review]

File: be/src/exec/pipeline/scan/olap_scan_context.h [hunk 2]
```diff
@@ -40,14 +42,52 @@ using OlapScanContextPtr = std::shared_ptr<OlapScanContext>;
...
 using OlapScanContextFactoryPtr = std::shared_ptr<OlapScanContextFactory>;
 
+class JitRewriterConcurrent {
+public:
+    JitRewriterConcurrent(int32_t dop) : _dop(dop), _barrier(), _errors(0), _id(0) {}
+    Status rewrite(std::vector<ExprContext*>& expr_ctxs, ObjectPool* pool, bool enable_jit);
+
+private:
+    // TODO: use c++20 barrier after upgrading gcc
+    class Barrier {
```
Comment (stdpain) [inline be/src/exec/pipeline/scan/olap_scan_context.h:52]: std::call_once is enough

File: be/src/exec/pipeline/scan/olap_scan_context.h [hunk 3]
```diff
@@ -40,14 +42,52 @@ using OlapScanContextPtr = std::shared_ptr<OlapScanContext>;
 class OlapScanContextFactory;
 using OlapScanContextFactoryPtr = std::shared_ptr<OlapScanContextFactory>;
 
+class JitRewriterConcurrent {
```
Comment (silverbullet233) [inline be/src/exec/pipeline/scan/olap_scan_context.h:45]: ConcurrentJitRewriter seems better?

File: be/src/exec/pipeline/scan/olap_scan_context.h (hunk 2 above)
Comment (fzhedu) [reply be/src/exec/pipeline/scan/olap_scan_context.h:52]: not really, if there are more than one expr, we can concurrently jit them, as jit may take tens of ms time.

**Review by silverbullet233 — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**